From abd92046f616ecfa15e2bcf90eb0528e34073f96 Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Sun, 23 Feb 2014 13:46:29 +0000
Subject: [PATCH 5/8] MusicCommon: handle TrackUnavailableEvents from the
 music player

* This sets the font state to 'disabled' for the tracks item in the
  "currentplaylist" button list. (This should be a common font state used
   elsewhere in the Watch Recordings screen for example)

* Also sets the "playstate" displaystate of the item to "unavailable"

These changes should allow themers to mirror in the currentplaylist buttonlist
what appears for not available recordings on the  Watch Recording screen.
---
 mythplugins/mythmusic/mythmusic/musiccommon.cpp |   23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/mythplugins/mythmusic/mythmusic/musiccommon.cpp b/mythplugins/mythmusic/mythmusic/musiccommon.cpp
index ac91ff1..b62f1f5 100644
--- a/mythplugins/mythmusic/mythmusic/musiccommon.cpp
+++ b/mythplugins/mythmusic/mythmusic/musiccommon.cpp
@@ -1688,6 +1688,29 @@ void MusicCommon::customEvent(QEvent *event)
         if (gPlayer->getCurrentMetadata() && trackID == gPlayer->getCurrentMetadata()->ID())
             updateTrackInfo(gPlayer->getCurrentMetadata());
     }
+    else if (event->type() == MusicPlayerEvent::TrackUnavailableEvent)
+    {
+        MusicPlayerEvent *mpe = dynamic_cast<MusicPlayerEvent *>(event);
+
+        if (!mpe)
+            return;
+
+        uint trackID = mpe->TrackID;
+
+        if (m_currentPlaylist)
+        {
+            for (int x = 0; x < m_currentPlaylist->GetCount(); x++)
+            {
+                MythUIButtonListItem *item = m_currentPlaylist->GetItemAt(x);
+                MusicMetadata *mdata = qVariantValue<MusicMetadata*> (item->GetData());
+                if (mdata && mdata->ID() == trackID)
+                {
+                    item->SetFontState("disabled");
+                    item->DisplayState("unavailable", "playstate");
+                }
+            }
+        }
+    }
 }
 
 void MusicCommon::updateVolume(void)
-- 
1.7.10

